staging-service/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/FileSystemDao.java

```diff
@@ -123,6 +125,11 @@ public void deleteBatch(final TenantId tenantId, final BatchId batchId)
         final Path inProgressBatchFolderPath = batchPathProvider.getInProgressPathForBatch(tenantId, batchId);
         final Path storageRefFolderPath = batchPathProvider.getStorageRefFolderPathForBatch(tenantId, batchId, this.storagePath, CONTENT_FILES);
         final List<String> fileNames = new ArrayList<>();
+        /*
+        Binary files must be sent before the json ones.
+        */
+        boolean jsonAlreadyRead = false;
```

Comment:
I think you've misunderstood.  All binary files don't need to be sent before all JSON files.  Referenced binary files need to be sent before JSON files that reference them.